website/src/eval-z3/eval-z3.ts

```diff
@@ -122,12 +122,18 @@ function compile(
 // - and strips types
 // successful output looks like `"use strict"; module.exports = (async () => { ... })();`
 export function compileZ3JS(src: string) {
+  // let imports = `
+  //   import type { init as initT, Model, Solver } from 'z3-solver';
+  //   declare let init: typeof initT;
+  //   declare let { Context }: Awaited<ReturnType<typeof init>>;
+  //   declare let Z3: ReturnType<typeof Context<'main'>>;
+  // `;
   let imports = `
-    import type { init as initT, Model, Solver } from 'z3-solver';
-    declare let init: typeof initT;
-    declare let { Context }: Awaited<ReturnType<typeof init>>;
-    declare let Z3: ReturnType<typeof Context<'main'>>;
-  `;
+  import type { init as initT, Model, Solver } from 'z3-solver';
+  declare let init: typeof initT;
+  declare let Z3: Awaited<ReturnType<typeof init>>;
+  declare let { Context, setParam } = Z3;
```

Comment:
@bakkot I'm not sure I'm on the right track, so definitely would appreciate your help here. The example failed to be compiled is
```js
// non-linear arithmetic

Z3.setParam('pp.decimal', true);
Z3.setParam('pp.decimal_precision', 20);

const x = Z3.Real.const('x');
const y = Z3.Real.const('y');
const z = Z3.Real.const('z');

const solver = new Z3.Solver();
solver.add(x.mul(x).add(y.mul(y)).eq(1)); // x^2 + y^2 == 1
solver.add(x.mul(x).mul(x).add(z.mul(z).mul(z)).lt('1/2')); // x^3 + z^3 < 1/2

await solver.check(); // sat
```

Reply:
Yeah, adding `declare let { Context, setParam } = Z3` should make the typechecker happy, and then including the change I suggested above should make the actual runtime behavior work.

Reply:
Not really, I got
```
Error: main.ts (-2,39): Initializers are not allowed in ambient contexts.
main.ts (6,14): Property 'Real' does not exist on type '{ em: any; Z3: { mk_context: (c: Z3_config) => Z3_context; mk_context_rc: (c: Z3_config) => Z3_context; global_param_set: (param_id: string, param_value: string) => void; ... 719 more ...; qe_lite: (c: Z3_context, vars: Z3_ast_vector, body: Z3_ast) => Z3_ast; }; } & Z3HighLevel'.
main.ts (7,14): Property 'Real' does not exist on type '{ em: any; Z3: { mk_context: (c: Z3_config) => Z3_context; mk_context_rc: (c: Z3_config) => Z3_context; global_param_set: (param_id: string, param_value: string) => void; ... 719 more ...; qe_lite: (c: Z3_context, vars: Z3_ast_vector, body: Z3_ast) => Z3_ast; }; } & Z3HighLevel'.
main.ts (8,14): Property 'Real' does not exist on type '{ em: any; Z3: { mk_context: (c: Z3_config) => Z3_context; mk_context_rc: (c: Z3_config) => Z3_context; global_param_set: (param_id: string, param_value: string) => void; ... 719 more ...; qe_lite: (c: Z3_context, vars: Z3_ast_vector, body: Z3_ast) => Z3_ast; }; } & Z3HighLevel'.
main.ts (10,23): Property 'Solver' does not exist on type '{ em: any; Z3: { mk_context: (c: Z3_config) => Z3_context; mk_context_rc: (c: Z3_config) => Z3_context; global_param_set: (param_id: string, param_value: string) => void; ... 719 more ...; qe_lite: (c: Z3_context, vars: Z3_ast_vector, body: Z3_ast) => Z3_ast; }; } & Z3HighLevel'.
    at runMicrotasks (<anonymous>)
```

Reply:
Oh, sorry, that's not legal syntax. Needs to be what it currently is, just with `setParam` also: `declare let { Context, setParam }: Awaited<ReturnType<typeof init>>;`

Reply:
I messed up with the types in this commit. After further understanding the high-level API the issue is now fixed. The imports should be
```
  declare let { Context, setParam }: Awaited<ReturnType<typeof init>>;
  declare let Z3: ReturnType<typeof Context<'main'>>;
```
simply add `setParam` to the destructure of `init()`